ionoscloud/data_source_cube_server.go

```diff
@@ -0,0 +1,446 @@
+				return err
+			}
+		}
+		if server.Entities != nil && server.Entities.Volumes != nil && server.Entities.Volumes.Items != nil && len(*server.Entities.Volumes.Items) > 0 &&
+			(*server.Entities.Volumes.Items)[0].Properties.Image != nil {
+			if err := d.Set("boot_image", *(*server.Entities.Volumes.Items)[0].Properties.Image); err != nil {
+				return err
+			}
+		}
+	}
+
+	if server.Entities == nil {
+		return nil
+	}
+
+	if server.Entities.Cdroms != nil && server.Entities.Cdroms.Items != nil && len(*server.Entities.Cdroms.Items) > 0 {
+		cdroms := setServerCDRoms(server.Entities.Cdroms.Items)
+		if err := d.Set("cdroms", cdroms); err != nil {
+			return err
+		}
+	}
+
+	var volumes []interface{}
+	if server.Entities.Volumes != nil && server.Entities.Volumes.Items != nil && len(*server.Entities.Volumes.Items) > 0 {
+		for _, volume := range *server.Entities.Volumes.Items {
+			entry := make(map[string]interface{})
+
+			entry["id"] = stringOrDefault(volume.Id, "")
+			entry["name"] = stringOrDefault(volume.Properties.Name, "")
+			entry["type"] = stringOrDefault(volume.Properties.Type, "")
+			//entry["size"] = float32OrDefault(volume.Properties.Size, 0)
+			entry["availability_zone"] = stringOrDefault(volume.Properties.AvailabilityZone, "")
+			entry["image_name"] = stringOrDefault(volume.Properties.Image, "")
+			entry["image_password"] = stringOrDefault(volume.Properties.ImagePassword, "")
+
+			if volume.Properties.SshKeys != nil && len(*volume.Properties.SshKeys) > 0 {
+				var sshKeys []interface{}
+				for _, sshKey := range *volume.Properties.SshKeys {
+					sshKeys = append(sshKeys, sshKey)
+				}
+				entry["ssh_keys"] = sshKeys
+			}
+
+			entry["bus"] = stringOrDefault(volume.Properties.Bus, "")
+			entry["licence_type"] = stringOrDefault(volume.Properties.LicenceType, "")
+			entry["cpu_hot_plug"] = boolOrDefault(volume.Properties.CpuHotPlug, true)
+			entry["ram_hot_plug"] = boolOrDefault(volume.Properties.RamHotPlug, true)
+			entry["nic_hot_plug"] = boolOrDefault(volume.Properties.NicHotPlug, true)
+			entry["nic_hot_unplug"] = boolOrDefault(volume.Properties.NicHotUnplug, true)
+			entry["disc_virtio_hot_plug"] = boolOrDefault(volume.Properties.DiscVirtioHotPlug, true)
+			entry["disc_virtio_hot_unplug"] = boolOrDefault(volume.Properties.DiscVirtioHotUnplug, true)
+			entry["device_number"] = int64OrDefault(volume.Properties.DeviceNumber, 0)
+			entry["pci_slot"] = int32OrDefault(volume.Properties.PciSlot, 0)
+			entry["backup_unit_id"] = stringOrDefault(volume.Properties.BackupunitId, "")
+			entry["user_data"] = stringOrDefault(volume.Properties.UserData, "")
+			entry["boot_server"] = stringOrDefault(volume.Properties.BootServer, "")
+
+			volumes = append(volumes, entry)
+		}
+
+		if err := d.Set("volumes", volumes); err != nil {
+			return err
+		}
+	}
+
+	var nics []interface{}
+	if server.Entities.Nics != nil && server.Entities.Nics.Items != nil && len(*server.Entities.Nics.Items) > 0 {
```

Comment:
make a func which returns []interface{} and use it here and to the server.go as well